tests/http/schema.py

```diff
@@ -1,18 +1,26 @@
-from typing import List, Optional
+from typing import IO, List, Optional
 
-from starlette.datastructures import UploadFile
+from starlette.datastructures import UploadFile as StarletteUploadFile
 
 import strawberry
 from strawberry.file_uploads import Upload
 from strawberry.permission import BasePermission
 from strawberry.types import Info
 
+from . import IS_STARLITE_INSTALLED
+
 
 def _read_file(text_file: Upload) -> str:
     # allow to keep this function synchronous, starlette's files have
     # async methods for reading
-    if isinstance(text_file, UploadFile):
+    text_file: IO
+    if isinstance(text_file, StarletteUploadFile):
         text_file = text_file.file._file  # type: ignore
+    if IS_STARLITE_INSTALLED:
```

Comment:
Looking at our `pyproject.toml`, we can rely on starlite being installed during testing. Pretty sure we can remove the `if IS_STARLITE_INSTALLED:` check.

Reply:
Starlite actually needs python ^3.8 and from my understanding of dependency spec it isn't installed in 3.7. As the `unit-tests` CI job runs the `tests/http` test suite on 3.7-3.11, I thought that we need a way to handle 3.7 where starlite isn't installed. Is that correct?

Reply:
Good point, you're right. I was thinking about whether there is a "better" way to handle this, but the only other option I came up with so far was something like this:

```python
try:
    from starlite import UploadFile as StarliteUploadFile
except ModuleNotFoundError:
    class StarliteUploadFile:
        pass

def _read_file(text_file: Upload) -> str:
    text_file: IO
    if isinstance(text_file, StarletteUploadFile):
        text_file = text_file.file._file  # type: ignore
    if isinstance(text_file, StarliteUploadFile):
        text_file = text_file
    content = text_file.read()
```

This approach for some reason felt more "standard" to me. Do you have any thoughts on it? Just wanted to share what I came up with thinking about it, not sure whether it's actually "better".

Reply:
Well, I don't have a strong opinion either way, but I think your snippet will require a bit more typing work to make mypy happy.